### PR TITLE
Unloading to a heap with a combine unloader

### PR DESCRIPTION
**Pull request by schwiti6190**
- Enabled unloading on a field for the combine unloader (#606):
  - Reverse unloading with an attached trailer
  - Side unload with a side tipping trailer
  - Auger wagon unload to a heap
- Reworked pipe offset mesaurement.
- Reworked settings code to enable cp field position settings.
- Enables Speeds greater than 50, if the vehicle allows it.
- Added pause button for field border recording.
- Fix for #2379 

Test:
- [x] Starting every mode
- [x] Pipe offset for self unload/combine unloader (towed harvesters)
- [x] MP sync of combine unloader/bunker silo parameters
- [x] Pause button field border recording
- [x] Check translation and help menu additions

### CODE REVIEW

File: translations/translation_ct.xml [hunk 1]
```diff
@@ -21,6 +21,7 @@
     <text name="CP_error_unloader_to_far_away_from_field" text="Unloader to far away from the field."/>
     <text name="CP_error_start_position_to_far_away_from_field" text="Starting position to far away from the field."/>
     <text name="CP_error_giants_unloader_not_available" text="Giants unloader not available."/>
+    <text name="CP_error_fieldUnloadPosition_to_far_away_from_field" text="Field unload position is to far away from the field."/>
```
Comment (pvaiko) [inline translations/translation_ct.xml:24]: Typo in text and name: to**o**

File: scripts/ai/AIDriveStrategyUnloadCombine.lua [hunk 2]
```diff
@@ -291,12 +326,30 @@ function AIDriveStrategyUnloadCombine:getDriveData(dt, vX, vY, vZ)
...
+    ---------------------------------------------
+    --- Unloading on the field
+    ---------------------------------------------
+    elseif self.state == self.states.DRIVE_TO_FIELD_UNLOAD_POSITION then
+        self:setMaxSpeed(self:getFieldSpeed())
+    elseif self.state == self.states.PREPARE_FOR_FIELD_UNLOAD then
+        self:prepareForFieldUnload()
+    elseif self.state == self.states.UNLOADING_ON_THE_FIELD then
+        self:setMaxSpeed(self.settings.reverseSpeed:getValue())
+        --- For some reason this is a problem ?
```
Comment (pvaiko) [inline scripts/ai/AIDriveStrategyUnloadCombine.lua:339]: Search for didNotMoveTimer in the code and comments, this shouldn't be necessary.

Comment (pvaiko) [reply scripts/ai/AIDriveStrategyUnloadCombine.lua:339]: Oh, I see you found it, still, it is a general CP problem, a generic solution would be better.

Comment (schwiti6190) [reply scripts/ai/AIDriveStrategyUnloadCombine.lua:338]: @pvaiko Should we fix this in another PR later?

Comment (pvaiko) [reply scripts/ai/AIDriveStrategyUnloadCombine.lua:338]: Sure.

**Pull request merged**